### ISSUE TITLE
Is CreationFunction is missing the `Promise<ImageryProvider>` type?

### ISSUE THREAD

**Issue by anne-gropler** [CesiumGS/cesium#11452]
The [`CreationFunction`](https://cesium.com/learn/cesiumjs/ref-doc/ProviderViewModel.html?classFilter=providerviewmo#.CreationFunction) is missing some types. The current definition is:

```
ImageryProvider|TerrainProvider|Array.<ImageryProvider>|Array.<TerrainProvider>|Promise.<TerrainProvider>|Promise.<Array.<TerrainProvider>>
```

However, I was recently updating to Version 1.106 where the imagery providers need to be created using `fromURL`.

Before, I could do this:

```ts
const providerViewModels = [] as Cesium.ProviderViewModel[];

providerViewModels.push(new Cesium.ProviderViewModel({
	name: 'Bing Maps Aerial',
	iconUrl: Cesium.buildModuleUrl('Widgets/Images/ImageryProviders/bingAerial.png'),
	tooltip: 'Bing Maps aerial imagery \nhttp://www.bing.com/maps',
	creationFunction: () => new Cesium.BingMapsImageryProvider({
		url: 'https://dev.virtualearth.net',
		key: bingMapsKey,
		mapStyle: Cesium.BingMapsStyle.AERIAL,
	}),
}));
```

Now this is not possible anymore, since the `BingMapsImageryProvider` constructor should not be used anymore. Instead, according to the docs, we need to use `from Url`.

```ts
providerViewModels.push(new Cesium.ProviderViewModel({
	name: 'Bing Maps Aerial',
	iconUrl: Cesium.buildModuleUrl('Widgets/Images/ImageryProviders/bingAerial.png'),
	tooltip: 'Bing Maps aerial imagery \nhttp://www.bing.com/maps',
	creationFunction: function() {
		return Cesium.BingMapsImageryProvider.fromUrl(
			'https://dev.virtualearth.net',
			{
				key: bingMapsKey,
				mapStyle: Cesium.BingMapsStyle.AERIAL,
			},
		) as any; // as any, because `Promise<ImageryProvider>` is not part of the `CreationFunction` type definition
	},
}));
```

Note the `as any`. If I don't put it there, I get the error below. To me, it looks like the `CreationFunction` type definition is just missing the  `Promise<ImageryProvider>` (and `Promise<ImageryProvider[]>` I suppose). Is it missing for a certain purpose, or was it forgotten?

I have this problem and found that [someone else on the community forum](https://community.cesium.com/t/creationfunction-in-providerviewmodel-mistype/25884/1) has the same issue.

```
Type '() => Promise<BingMapsImageryProvider>' is not assignable to type 'CreationFunction | Command'.
  Type '() => Promise<BingMapsImageryProvider>' is not assignable to type 'CreationFunction'.
    Type 'Promise<BingMapsImageryProvider>' is not assignable to type 'TerrainProvider | ImageryProvider | ImageryProvider[] | TerrainProvider[] | Promise<TerrainProvider> | Promise<...>'.
      Type 'Promise<BingMapsImageryProvider>' is not assignable to type 'Promise<TerrainProvider>'.
        Type 'BingMapsImageryProvider' is missing the following properties from type 'TerrainProvider': hasWaterMask, hasVertexNormals, availability, requestTileGeometry, and 3 more.ts(2322)
Cesium.d.ts(43996, 9): The expected type comes from property 'creationFunction' which is declared here on type '{ name: string; tooltip: string; iconUrl: string; category?: string; creationFunction: CreationFunction | Command; }'
```

**Comment by Oceanestars** [comment]
Thanks for writing the issue! To answer your question from the community form I did try to cast it with as Imageryprovider and other type. I also tried to use one from the Cesium code(which I think would be the right type) with no luck.

I used this [testProvider](https://github.com/CesiumGS/cesium/blob/55952193759056f5ec6d13b68cbd6d6fcd4f21c3/packages/widgets/Specs/Viewer/ViewerSpec.js#L59C21-L59C21) to check the types and had the same errors as above
